src/dynamic_boundary_conditions/rain_depth_data_from_hirds.py

```diff
@@ -7,11 +7,14 @@
 
 import requests
 from requests.structures import CaseInsensitiveDict
+import re
 import pandas as pd
+import pathlib
 
 
-def get_url_id(site_id: str) -> str:
-    """Each site has a unique key that need to be inserted in the url before making an api request."""
+def get_site_url_key(site_id: str, idf: bool) -> str:
+    """Get each sites' unique url key from the hirds website using curl commands.
+    idf: set to False to get rainfall depth data, and True to get rainfall intensity data."""
```

Comment:
What are your thoughts @caseyli92 on having one function for each `idf` setting? This way works fine so we can keep it but what are your thoughts? Too much effort? Messier solution with more duplication?

Reply:
At this stage, I think it will take a bit of effort to revert the setting of `īdf` parameters in all related functions. Also, I think it would be a messier solution with more duplications. So it would probably be better to keep it in its current state.